ThingiBrowser/PreferencesHelper.py

```diff
@@ -23,22 +23,31 @@ def initSetting(cls, setting_name: str, default_value: Optional[str] = "") -> st
         return preferences.getValue(preference_key)
 
     @classmethod
-    def getAllSettings(cls) -> List[Dict[str, str]]:
+    def getAllSettings(cls) -> List[Dict[str, Union[str, List[Dict[str, str]]]]]:
         """
         Get all settings as key:value dict.
         # FIXME: convert to QtListModel.
         :return: The settings dict.
         """
         return [
             {
+                "type": "text",
                 "key": Settings.THINGIVERSE_USER_NAME_PREFERENCES_KEY,
                 "value": cls.getSettingValue(Settings.THINGIVERSE_USER_NAME_PREFERENCES_KEY),
                 "label": "Thingiverse username"
             },
             {
+                "type": "text",
                 "key": Settings.MYMINIFACTORY_USER_NAME_PREFERENCES_KEY,
                 "value": cls.getSettingValue(Settings.MYMINIFACTORY_USER_NAME_PREFERENCES_KEY),
                 "label": "MyMiniFactory username"
+            },
+            {
+                "type": "combobox",
+                "key": Settings.DEFAULT_API_CLIENT,
+                "value": cls.getSettingValue(Settings.DEFAULT_API_CLIENT),
+                "label": "Default API service",
```

Comment:
For user friendlyness, maybe rename this label to "Default repository service" or something.